### PR TITLE
fix(pii): Fix scrubbing user paths in minidump debug modules

### PR DESCRIPTION
**Pull request by jjbayer**
https://github.com/getsentry/relay/pull/2724 silently broke user path scrubbing in minidumps. The regression was not automatically flagged because minidump scrubbing [ignores regexes that cannot be compiled with unicode disabled](https://github.com/getsentry/relay/blob/63ca425f5eba144cd5b8976b4201cd61084dafb9/relay-pii/src/attachments.rs#L37-L41), regardless of whether they are user-defined regexes or builtin static ones.

This PR both fixes the regex and adds an automatic test to each PII regex, so we can catch similar problems in the future.

### CODE REVIEW

File: relay-pii/src/regexes.rs [hunk 1]
```diff
@@ -106,186 +106,239 @@ macro_rules! ip {
...
+            #[test]
+            fn supports_byte_mode() {
+                assert!(regex::bytes::RegexBuilder::new($name.as_str())
+                    .unicode(false)
+                    .multi_line(false)
+                    .dot_matches_new_line(true)
+                    .build()
+                    .is_ok());
+            }
+        }
```
Comment (jjbayer) [inline relay-pii/src/regexes.rs:126]: This makes sure the regex can also be used by minidump scrubbing. There are several open questions (see below), but this was the most conservative approach I could think of.

1. Why does minidump scrubbing need `.unicode(false)`? https://github.com/getsentry/relay/blob/b5b738f4b2c9a546acaa67fb80026fe46bfadb41/relay-pii/src/attachments.rs#L29-L31
2. Could we automatically generate a non-unicode version of every regex, such that it does not have to be recompiled on the fly for every minidump?

File: relay-pii/src/regexes.rs [hunk 2]
```diff
@@ -106,186 +106,239 @@ macro_rules! ip {
...
-            (
-                [A–Za-z0–9'\.\ _!\#^~-]+
-            )
-        ",
-    )
-    .unwrap()
-});
+        )
+        (
+            [^/\\\r\n]+
```
Comment (jjbayer) [inline relay-pii/src/regexes.rs:280]: This is the change that fixes the regex with unicode disabled, while still meeting the requirements of https://github.com/getsentry/relay/pull/2724: https://github.com/getsentry/relay/blob/b5b738f4b2c9a546acaa67fb80026fe46bfadb41/relay-pii/src/builtin.rs#L1247-L1248

File: relay-pii/src/regexes.rs [hunk 3]
```diff
@@ -106,186 +106,239 @@ macro_rules! ip {
     (v6s) => { "[0-9a-fA-F]{1,4}" };
 }
 
+macro_rules! regex {
+    ($name:ident, $rule:expr) => {
+        #[allow(non_snake_case)]
+        mod $name {
+            use super::*;
+            pub static $name: Lazy<Regex> = Lazy::new(|| Regex::new($rule).unwrap());
```
Comment (Dav1dde) [inline relay-pii/src/regexes.rs]: Could take the opportunity and switch this to `std::sync::LazyLock` (or we do that separately).

File: relay-pii/src/regexes.rs (hunk 1 above)
Comment (Dav1dde) [reply relay-pii/src/regexes.rs:126]: 1. I read the issue and I still don't understand why this is necessary, would be nice if we had a test for this...
2. That'd be nice or if we didn't even need the special casing in the first place.

**Review by Dav1dde — APPROVED** [review]

**Pull request merged**